Tests/Opc.Ua.Security.Certificates.Tests/CRLTests.cs

```diff
@@ -0,0 +1,188 @@
+ * obtaining a copy of this software and associated documentation
+ * files (the "Software"), to deal in the Software without
+ * restriction, including without limitation the rights to use,
+ * copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the
+ * Software is furnished to do so, subject to the following
+ * conditions:
+ * 
+ * The above copyright notice and this permission notice shall be
+ * included in all copies or substantial portions of the Software.
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES
+ * OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,
+ * WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+ * OTHER DEALINGS IN THE SOFTWARE.
+ *
+ * The complete license agreement can be found here:
+ * http://opcfoundation.org/License/MIT/1.00/
+ * ======================================================================*/
+
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Security.Cryptography;
+using System.Security.Cryptography.X509Certificates;
+using System.Text;
+using NUnit.Framework;
+
+namespace Opc.Ua.Security.Certificates.Tests
+{
+    /// <summary>
+    /// Tests for the CRL class.
+    /// </summary>
+    [TestFixture, Category("CRL")]
+    [Parallelizable]
+    [SetCulture("en-us")]
+    public class CRLTests
+    {
+        #region DataPointSources
+        [DatapointSource]
+        public CRLAsset[] CRLTestCases = new AssetCollection<CRLAsset>(TestUtils.EnumerateTestAssets("*.crl")).ToArray();
+        #endregion
+
+        #region Test Setup
+        /// <summary>
+        /// Set up a Global Discovery Server and Client instance and connect the session
+        /// </summary>
+        [OneTimeSetUp]
+        protected void OneTimeSetUp()
+        {
+        }
+
+        /// <summary>
+        /// Clean up the Test PKI folder
+        /// </summary>
+        [OneTimeTearDown]
+        protected void OneTimeTearDown()
+        {
+        }
+        #endregion
+
+        #region Test Methods
+        /// <summary>
+        /// Verify self signed app certs.
+        /// </summary>
+        [Theory]
+        public void DecodeCRLs(
+            CRLAsset crlAsset
+            )
+        {
+            var x509Crl = new X509CRL(crlAsset.Crl);
+            Assert.NotNull(x509Crl);
+            TestContext.Out.WriteLine($"CRLAsset:   {x509Crl.Issuer}");
+            var crlInfo = WriteCRL(x509Crl);
+            TestContext.Out.WriteLine(crlInfo);
+        }
+
+        /// <summary>
+        /// Validate a CRL Builder and decoder pass.
+        /// </summary>
+        [Test]
+        public void CrlInternalBuilderTest()
```

Comment:
what is the difference to test `CrlBuilderTest` they seem to be identically

Reply:
not anymore. One is just testiong the internal encoder/decoder, the other *now* with signing